.github/workflows/tests.yml

```diff
@@ -0,0 +1,35 @@
+name: Tests
+on: [push, pull_request, workflow_dispatch] #workflow_dispatch works only if its active in the main branch
```

Comment:
Does this mean this workflow runs every time there is a push to this branch? Should we expect the readme to be updated with the latest numbers whenever there is a push?

Reply:
the README will just show the latest coverage badge for the `main` branch. Markdown isn't sophisticated enough to parameterize the badge image URL and render a different badge for different branches. But clicking on the badge links out to the full coverage report across all branches on codecov